src/main.js

```diff
@@ -0,0 +1,21 @@
+import { createCharacterCardComponent } from "./lib/characterCard.js";
+import { fetchCharacterCards } from "./lib/api.js";
+
+const searchButton = document.querySelector(".header__search-button");
+const filterDropdown = document.querySelector(".header__filter");
+const main = document.querySelector(".main");
+
+searchButton.addEventListener("click", () => {
+  const status = filterDropdown.value;
+  clearCharacterCards();
+  fetchCharacterCards(status).then((characterCards) => {
```

Comment:
maybe you can move line 11 - 13 (starting at .then) into the fetchCharacterCards function, to have a cleaner code here in the main.js